readme
dedenie - fyzickaOsoba:9
zapuzdrenie - Controller:44
agregacia - user:15
Rozhranie - platcaDPH
organizacia kodu - baliky MVC
navrhovy vzor - Strategy
Default methods - platenieDPH:12
GUI - View (oddelene)
Multithreading - View:42
Generickost - fyzickaOsoba:40
RTTI - fyzickaOsoba:40
Lambda vyrazy - Controller:459

changelog.txt
commit 7b528115ca8ca4a9a58586f44e1e07103a5c76e1
Author: vecnehladny <vecnehladny@vecnehladnys-MacBook-Pro.local>
Date:   Sat May 11 19:09:15 2019 +0200

    update diagramu tried

commit e3c3d88c6ab0953e76d2b74b79e2fb2f36f58fd5
Author: vecnehladny <vecnehladny@vecnehladnys-MacBook-Pro.local>
Date:   Sat May 11 18:34:27 2019 +0200

    pridany javadoc + upravene readme

commit ce327ad96ae495858c060a5915f441eebb0443ee
Author: vecnehladny <vecnehladny@vecnehladnys-MacBook-Pro.local>
Date:   Sat May 11 17:36:14 2019 +0200

    vypocet dph update

commit 30e4311362d0b693061ef5b19289e33556d18a65
Author: vecnehladny <vecnehladny@vecnehladnys-MacBook-Pro.local>
Date:   Sat May 11 17:27:27 2019 +0200

    pridanie navrhoveho vzoru Strategy

commit f85da8603ceb41ceb8b966c6be92c09e010d8f98
Author: vecnehladny <vecnehladny@vecnehladnys-MacBook-Pro.local>
Date:   Thu Apr 25 13:15:22 2019 +0200

    pridany polymorfizmus na vypocitanie dph pre platcov

commit 6b3dc73e8f4c9ba0b5a4222ec32351ad88b1e369
Author: vecnehladny <vecnehladny@vecnehladnys-MacBook-Pro.local>
Date:   Thu Apr 25 12:35:15 2019 +0200

    upravene readme

commit f8afb17c2482ab3a20e686aec948e0734168b2ac
Author: vecnehladny <vecnehladny@vecnehladnys-MacBook-Pro.local>
Date:   Thu Apr 25 12:29:04 2019 +0200

    pridany diagram

commit 24e855979ae2c1a01e7964eb87308120c3e030f7
Author: vecnehladny <vecnehladny@vecnehladnys-MacBook-Pro.local>
Date:   Thu Apr 25 12:13:09 2019 +0200

    opraveny bug s nevypisovanymim odovzdanych danovych priznani

commit ad5c95e62528539ccdc8663e6f6cb6dade3794a0
Author: vecnehladny <vecnehladny@vecnehladnys-MacBook-Pro.local>
Date:   Sun Apr 21 17:46:34 2019 +0200

    opraveny bug s vymazavanim uzivatela

commit c877d64c13fa1bc25ee38fd6efcf4ece2bb29913
Author: vecnehladny <viktorklima@yahoo.com>
Date:   Tue Apr 9 03:03:05 2019 +0200

    removed submodule

commit 2eb15c30eb90e7e78a4e64c10ceea09edf88e92a
Author: vecnehladny <viktorklima@yahoo.com>
Date:   Tue Apr 9 02:45:49 2019 +0200

    Revert "Merge remote-tracking branch 'sub/master'"
    
    This reverts commit 5e2d610e206e2cb7b280d2f2a5c390e0bc1adbc5, reversing
    changes made to 7856796877533f408397726fd6ee028c6ef48635.

commit 5e2d610e206e2cb7b280d2f2a5c390e0bc1adbc5
Merge: 7856796 a18b63d
Author: vecnehladny <viktorklima@yahoo.com>
Date:   Tue Apr 9 02:44:44 2019 +0200

    Merge remote-tracking branch 'sub/master'
    
    Refaktoring

commit 7856796877533f408397726fd6ee028c6ef48635
Author: vecnehladny <viktorklima@yahoo.com>
Date:   Tue Apr 9 02:38:24 2019 +0200

    refaktoring

commit a18b63d3a4aa8c8cd773e694730f138eb514718d
Author: vecnehladny <viktorklima@yahoo.com>
Date:   Tue Apr 9 02:27:59 2019 +0200

    refaktoring

commit eb09bc1b4744f02b7a5a9658dac350ffed6b59a0
Author: vecnehladny <viktorklima@yahoo.com>
Date:   Tue Apr 9 02:05:30 2019 +0200

    serializacia

commit b6ba38a60537a32f79f9675766b8aeb0b8a2fe57
Author: vecnehladny <viktorklima@yahoo.com>
Date:   Tue Apr 9 02:02:32 2019 +0200

    Refaktoring

commit 2614f5b80b5f234802144bc6ada491163d9c33ce
Author: vecnehladny <viktorklima@yahoo.com>
Date:   Tue Apr 9 01:19:32 2019 +0200

    Revert "Revert "presunutie repozitara""
    
    This reverts commit 273866dd1894b87078aa82c5f1f30a23008aab7e.

commit 273866dd1894b87078aa82c5f1f30a23008aab7e
Author: vecnehladny <viktorklima@yahoo.com>
Date:   Tue Apr 9 01:10:27 2019 +0200

    Revert "presunutie repozitara"
    
    This reverts commit fbb5e568c39474118889b0b360f9642cce8cbbcc.

commit fbb5e568c39474118889b0b360f9642cce8cbbcc
Author: vecnehladny <viktorklima@yahoo.com>
Date:   Tue Apr 9 01:08:44 2019 +0200

    presunutie repozitara

commit 6230d5035b1f69090823bc570c5bee590bd8df17
Author: vecnehladny <viktorklima@yahoo.com>
Date:   Tue Apr 9 01:01:28 2019 +0200

    pridane pridavanie danovych priznani, vypis v profile uzivatela

commit cddab1685fb33d3bca93ce88d9927af70c10a848
Author: vecnehladny <viktorklima@yahoo.com>
Date:   Mon Apr 8 12:39:15 2019 +0200

    pridane atributy danoveho priznania do classy

commit 1070c39ed0c3b4ddcc02f06bebe44377e05f1c3b
Author: vecnehladny <viktorklima@yahoo.com>
Date:   Mon Apr 8 01:00:52 2019 +0200

    zacate pridavanie atributov danoveho priznania

commit ea79a14442a17d20e4828a14ef6bb291f3995027
Author: vecnehladny <viktorklima@yahoo.com>
Date:   Mon Apr 8 00:46:30 2019 +0200

    pridane fx:id pre cely dokument danoveho priznania, zaciatok handlovania dat z danoveho priznania

commit 37ffd0bf85da2797b6e0a158229df43f39721a35
Author: vecnehladny <viktorklima@yahoo.com>
Date:   Sun Apr 7 23:21:43 2019 +0200

    pridana 6. strana sceny danoveho priznania + upravene css

commit 12e76bc16e522316639f97419f1e8abf1404748b
Author: vecnehladny <viktorklima@yahoo.com>
Date:   Sun Apr 7 22:15:38 2019 +0200

    pridana 5. strana sceny danoveho priznania

commit c16b6cf3ea848361c4debcbb283d383717561895
Author: vecnehladny <viktorklima@yahoo.com>
Date:   Sun Apr 7 19:46:21 2019 +0200

    pridana 4, strana sceny danoveho priznania

commit 7dd6b9a560b53c0fbf8ec862fbcdaaf603b8cfaa
Author: vecnehladny <viktorklima@yahoo.com>
Date:   Sun Apr 7 19:17:28 2019 +0200

    pridana 3. strana sceny danoveho priznania

commit b5b21d63442b4dee7c480ec0f1be781c0c4eb740
Author: vecnehladny <viktorklima@yahoo.com>
Date:   Sun Apr 7 18:42:24 2019 +0200

    pridana 2. strana sceny danoveho priznania

commit 1bc5d93ec074533275ec4890d0d4b9d4345aaa4c
Author: vecnehladny <viktorklima@yahoo.com>
Date:   Sun Apr 7 14:33:02 2019 +0200

    update sceny danoveho priznania + css

commit 605d1fbc1722568399c155bcb17b1e22cf9080c8
Author: vecnehladny <viktorklima@yahoo.com>
Date:   Sun Apr 7 13:18:47 2019 +0200

    pridanie tried danovych priznani, vytvorenie sceny danoveho priznania

commit b0a936886cc23299e00a322f3ea6dc6beb1bfc96
Author: vecnehladny <viktorklima@yahoo.com>
Date:   Fri Apr 5 19:23:38 2019 +0200

    pridana funkcionalita pridania subjekta do databazy

commit 36d21b4171c369581c8b201d0e4e51c1f4bf4b9b

application/Database.java
package application;

import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.ObjectInputStream;
import java.io.ObjectOutputStream;
import java.io.Serializable;
import java.util.ArrayList;

import users.fyzickaOsoba;
import users.pravnickaOsoba;
import users.user;

public class Database implements Serializable {
	private static ArrayList<user> subjekti = new ArrayList<user>();

	
	public static ArrayList<user> search(String key) {
		ArrayList<user> found = new ArrayList<user>();
		
		for(user user : subjekti) {
			if(user.getMeno().equalsIgnoreCase(key)) {
				found.add(user);
			}
			
			if(user.getDIC().equalsIgnoreCase(key)) {
				found.add(user);
			}
			
			if(user.getICO().equalsIgnoreCase(key)) {
				found.add(user);
			}
		}
		
		System.out.println(found);
		return found;
	}
	
	public static ArrayList<user> getSubjekti() {
		return subjekti;
	}
	
	public static void save() {
		try {
			FileOutputStream fileOut = new FileOutputStream("databaza.ser");
			ObjectOutputStream out = new ObjectOutputStream(fileOut);
			
			out.writeObject(subjekti);
			
			out.close();
			fileOut.close();
		}
		
		catch(FileNotFoundException e) {
			e.printStackTrace();
		}
		
		catch(IOException e) {
			e.printStackTrace();
		}
	}
	
	public static void load() {
		subjekti.clear();
		
		try{
			FileInputStream fileIn = new FileInputStream("databaza.ser");
            ObjectInputStream in = new ObjectInputStream(fileIn);
            
			subjekti = (ArrayList<user>)in.readObject();
				
			for(user u : subjekti) {
				System.out.println("Recovered " + u.getMeno() + " " + u.getICO());
			}
			
			in.close();
            fileIn.close();
            
		}
		
		catch (ClassNotFoundException e) {
            e.printStackTrace();
        } 
		
		catch(FileNotFoundException e) {
            e.printStackTrace();
        }
		
		catch (IOException e) {
            e.printStackTrace();
        }
		
	}
	
	public static void add(user u) {
		subjekti.add(u);
		save();
		load();
	}
	
	public static void remove(String ico) {
		for(user u : subjekti) {
			if(u.getICO().equalsIgnoreCase(ico)) {
				System.out.println(u);
				subjekti.remove(u);
			}
		}
		save();
		load();
	}
	
	public void create() {
		fyzickaOsoba a = new fyzickaOsoba("Janko", 30, "Muž", "Budatinska 22", "Bratislava", "97901", "13957971", "7096487980", true);
		fyzickaOsoba b = new fyzickaOsoba("Jozef", 30, "Muž", "Budatinska 23", "Bratislava", "97901", "88662143", "7116725051", true);
		fyzickaOsoba c = new fyzickaOsoba("Peter", 30, "Muž", "Budatinska 24", "Bratislava", "97901", "24962304", "8935605926", false);
		fyzickaOsoba d = new fyzickaOsoba("Mariana", 30, "Žena", "Budatinska 25", "Bratislava", "97901", "90112069", "6823592351", true);
		fyzickaOsoba e = new fyzickaOsoba("Jaroslav", 30, "Muž", "Budatinska 26", "Bratislava", "97901", "10458637", "6608787861", true);
		fyzickaOsoba f = new fyzickaOsoba("Rasto", 30, "Muž", "Budatinska 27", "Bratislava", "97901", "19608847", "1412213214", true);
		fyzickaOsoba g = new fyzickaOsoba("Mikulas", 30, "Muž", "Budatinska 28", "Bratislava", "97901", "66455841", "6133345446", false);
		fyzickaOsoba h = new fyzickaOsoba("Demeter", 30, "Muž", "Budatinska 29", "Bratislava", "97901", "73655436", "4970341139", false);
		fyzickaOsoba i = new fyzickaOsoba("Stefania", 30, "Žena", "Budatinska 30", "Bratislava", "97901", "23579430", "6410762553", false);
		fyzickaOsoba j = new fyzickaOsoba("Pavlina", 30, "Žena", "Budatinska 31", "Bratislava", "97901", "45148532", "4259677712", true);
		
		pravnickaOsoba k = new pravnickaOsoba("Janko", 30, "Muž", "Budatinska 32", "Bratislava", "97901", "37182592", "2876954278", true);
		pravnickaOsoba l = new pravnickaOsoba("Jozef", 30, "Muž", "Budatinska 33", "Bratislava", "97901", "71022110", "8057768850", true);
		pravnickaOsoba m = new pravnickaOsoba("Peter", 30, "Muž", "Budatinska 34", "Bratislava", "97901", "58669440", "6953943740", false);
		pravnickaOsoba n = new pravnickaOsoba("Mariana", 30, "Žena", "Budatinska 35", "Bratislava", "97901", "36131386", "8482091457", true);
		pravnickaOsoba o = new pravnickaOsoba("Jaroslav", 30, "Muž", "Budatinska 36", "Bratislava", "97901", "18540186", "6175742984", true);
		pravnickaOsoba p = new pravnickaOsoba("Rasto", 30, "Muž", "Budatinska 37", "Bratislava", "97901", "45313930", "5855533601", true);
		pravnickaOsoba q = new pravnickaOsoba("Mikulas", 30, "Muž", "Budatinska 38", "Bratislava", "97901", "47380567", "5361955783", false);
		pravnickaOsoba r = new pravnickaOsoba("Demeter", 30, "Muž", "Budatinska 39", "Bratislava", "97901", "30921635", "4729612918", false);
		pravnickaOsoba s = new pravnickaOsoba("Stefania", 30, "Žena", "Budatinska 40", "Bratislava", "97901", "95730474", "2298889872", false);
		pravnickaOsoba t = new pravnickaOsoba("Pavlina", 30, "Žena", "Budatinska 41", "Bratislava", "97901", "54106635", "6297867825", true);
		
		subjekti.add(a);
		subjekti.add(b);
		subjekti.add(c);
		subjekti.add(d);
		subjekti.add(e);
		subjekti.add(f);
		subjekti.add(g);
		subjekti.add(h);
		subjekti.add(i);
		subjekti.add(j);
		subjekti.add(k);
		subjekti.add(l);
		subjekti.add(m);
		subjekti.add(n);
		subjekti.add(o);
		subjekti.add(p);
		subjekti.add(q);
		subjekti.add(r);
		subjekti.add(s);
		subjekti.add(t);
		System.out.println("databaza vytvore");
	}
	
	public Database() {
		//create();
		//save();
		load();
	}
}
